src/Nlog.RabbitMQ.Target/MessageFormatter.cs

```diff
@@ -47,7 +47,20 @@ public static string GetMessageInner(JsonSerializer jsonSerializer, string mesag
 
                     logLine.AddField(key, propertyPair.Value);
                 }
-            }
+            }
+			
+            if (addGdc)
+                foreach (var gdc in GlobalDiagnosticsContext.GetNames())
+                    logLine.AddField(gdc, GlobalDiagnosticsContext.GetObject(gdc));
+
+            if (addNdlc)
+                foreach (var ndlc in NestedDiagnosticsLogicalContext.GetAllObjects())
+                    logLine.AddField("nested", ndlc);
+
+            if (addMdlc)
+                foreach (var mdlc in MappedDiagnosticsLogicalContext.GetNames())
```

Comment:
This will not work when RabbitMQTarget  is used together with `<targets async="true">`. This is why `TargetWithContext` was introduced.

Reply:
It works with GetAllProperties(logEvent). I'll replace it with a simple foreach over it.